### PR TITLE
Merge 3.3.0 reference

### PR DESCRIPTION
**Pull request by Kordyjan**

### CODE REVIEW

File: docs/_docs/reference/contextual/derivation.md
Comment (julienrf) [inline docs/_docs/reference/contextual/derivation.md:1]: I am not sure about the changes in this file. @bishabosha could you please double check?

Comment (bishabosha) [reply docs/_docs/reference/contextual/derivation.md:1]: this change is ok, (all I see is h3 changing to h2)

**Review by prolativ — APPROVED** [review]

**Review by bishabosha — CHANGES_REQUESTED** [review]
we really should update https://github.com/lampepfl/dotty/blob/release-3.3.0/docs/_docs/reference/language-versions/source-compatibility.md to list `-source 3.3` mode

**Comment by Kordyjan** [comment]
I don't think we have a version picker for the language reference. It exists only for API docs.
Language ref is always synchronized with `language-reference-stable`.

File: docs/_docs/reference/language-versions/source-compatibility.md [hunk 1]
```diff
@@ -23,6 +23,9 @@ The default Scala language syntax version currently supported by the Dotty compi
...
 - [`future`](https://scala-lang.org/api/3.x/scala/runtime/stdLibPatches/language$$future$.html): A preview of changes that will be introduced in `3.x` versions after `3.2`.
 Some Scala 2 specific idioms are dropped in this version. The feature set supported by this version may grow over time as features become stabilised for preview.
+- [`3.3`](https://scala-lang.org/api/3.x/scala/runtime/stdLibPatches/language$$3/3$.html): the same as `3.2`, but in addition:
+  -[Fewer braces syntax](https://docs.scala-lang.org/scala3/reference/other-new-features/indentation.html#optional-braces-for-method-arguments-1) is enabled by default.
+- [`3.3-migration`](https://scala-lang.org/api/3.x/scala/runtime/stdLibPatches/language$$3/3-migration$.html): the same as `3.3`
 
 - [`future-migration`](https://scala-lang.org/api/3.x/scala/runtime/stdLibPatches/language$$future-migration$.html): Same as `future` but with additional helpers to migrate from `3.2`. Similarly to the helpers available under `3.0-migration`, these include migration warnings and optional rewrites.
```
Comment (bishabosha) [inline docs/_docs/reference/language-versions/source-compatibility.md:30]:
```suggestion
- [`future-migration`](https://scala-lang.org/api/3.x/scala/runtime/stdLibPatches/language$$future-migration$.html): Same as `future` but with additional helpers to migrate from `3.3`. Similarly to the helpers available under `3.0-migration`, these include migration warnings and optional rewrites.
```

**Pull request merged**